test/test_elastic_retriever.py

```diff
@@ -33,3 +37,49 @@ def test_elasticsearch_retrieval_filters(retriever_with_docs, document_store_wit
 
     res = retriever_with_docs.retrieve(query="Who lives in Berlin?", filters={"name":["filename1"], "meta_field":["test2"]})
     assert len(res) == 0
+
+
+@pytest.mark.elasticsearch
+def test_elasticsearch_custom_query(elasticsearch_fixture):
+    client = Elasticsearch()
+    client.indices.delete(index='haystack_test_custom', ignore=[404])
+    document_store = ElasticsearchDocumentStore(index="haystack_test_custom", text_field="custom_text_field",
+                                                embedding_field="custom_embedding_field")
+    documents = [
+        {"text": "test_1", "meta": {"year": "2019"}},
+        {"text": "test_2", "meta": {"year": "2020"}},
+        {"text": "test_3", "meta": {"year": "2021"}},
+        {"text": "test_4", "meta": {"year": "2021"}},
+        {"text": "test_5", "meta": {"year": "2021"}},
+    ]
+    document_store.write_documents(documents)
+
+    # test custom "terms" query
+    retriever = ElasticsearchRetriever(
+        document_store=document_store,
+        custom_query="""
+            {
+                "size": 10, 
+                "query": {
+                    "bool": {
+                        "should": [{
+                            "multi_match": {"query": "${query}", "type": "most_fields", "fields": ["text"]}}],
+                            "filter": [{"terms": {"year": ${years}}}]}}}"""
+    )
+    results = retriever.run(query="test", filters={"years": ["2020", "2021"]})[0]["documents"]
+    assert len(results) == 4
+
+    # test custom "term" query
+    retriever = ElasticsearchRetriever(
+        document_store=document_store,
+        custom_query="""
+                {
+                    "size": 10, 
+                    "query": {
+                        "bool": {
+                            "should": [{
+                                "multi_match": {"query": "${query}", "type": "most_fields", "fields": ["text"]}}],
```

Comment:
Why is it `"${query}"`here but one line afterwards it's only `${years}` without quotation marks?

Reply:
Yes, that inconsistent. I removed quotes around the `query` placeholder as well.